elastic/kibana
# [lens] show 'View details' UI action to open clusters inspector tab when request fails

Closes https://github.com/elastic/kibana/issues/171570

PR make the following changes
1. Consolidates data EsError logic and Lens EsError logic. This resulted in being able to remove large parts of lens error_helper.tsx file.
2. Consolidates lens WorkspacePanel error logic. Before PR configuration errors and data loading errors each rendered their own version of EuiEmptyPrompt with slightly different behavior. Now, both render WorkspaceErrors component and have the same behavior
3. Updated lens ExpressionWrapper to return original error to embeddable output.

### Test - EsError in embeddable
1. install sample web logs
2. create new dashboard
3. Click "Create visualization"
4. Drag "timestamp" field into workspace.
5. Click "Save and return"
6. Add filter
    ```
    {
      "error_query": {
        "indices": [
          {
            "error_type": "exception",
            "message": "local shard failure message 123",
            "name": "kibana_sample_data_logs"
          }
        ]
      }
    }
    ```
7. Verify EsError and "View details" action are displayed
    <img width="500" alt="Screenshot 2023-12-08 at 1 34 20 PM" src="https://github.com/elastic/kibana/assets/373691/1c65b7f3-ece7-4000-a5b2-11127cc38f01">

### Test - multiple configuration errors in lens editor
1. install sample web logs
2. create new lens visualization
3. Drag "timestamp" field into workspace.
4. Add filter
    ```
    {
      "error_query": {
        "indices": [
          {
            "error_type": "exception",
            "message": "local shard failure message 123",
            "name": "kibana_sample_data_logs"
          }
        ]
      }
    }
    ```
5. Verify EsError and "View details" action are displayed
    <img width="500" alt="Screenshot 2023-12-08 at 1 09 22 PM" src="https://github.com/elastic/kibana/assets/373691/22023512-d344-4f99-abbd-8427d764f821">

### Test - EsError in embeddable
1. install sample web logs
2. create new dashboard
3. Click "Create visualization"
4. Drag "timestamp" field into workspace.
5. Change "Vertical axis" to "Cumulative sum". Select "Field" select and hit delete key
6. Clone layer one or more times
7. Verify pagination is displayed, allowing users to click through all configuration errors
    <img width="500" alt="Screenshot 2023-12-08 at 12 59 18 PM" src="https://github.com/elastic/kibana/assets/373691/6302658a-8cf7-4a1a-a117-ae810c0af539">
